store/gaskvstore.go

```diff
@@ -26,83 +28,95 @@ func NewGasKVStore(gasMeter sdk.GasMeter, gasConfig sdk.GasConfig, parent sdk.KV
 }
 
 // Implements Store.
-func (gi *gasKVStore) GetStoreType() sdk.StoreType {
-	return gi.parent.GetStoreType()
+func (gs *gasKVStore) GetStoreType() sdk.StoreType {
+	return gs.parent.GetStoreType()
 }
 
 // Implements KVStore.
-func (gi *gasKVStore) Get(key []byte) (value []byte) {
-	gi.gasMeter.ConsumeGas(gi.gasConfig.ReadCostFlat, "ReadFlat")
-	value = gi.parent.Get(key)
+func (gs *gasKVStore) Get(key []byte) (value []byte) {
+	gs.gasMeter.ConsumeGas(gs.gasConfig.ReadCostFlat, sdk.GasReadCostFlatDesc)
+	value = gs.parent.Get(key)
+
 	// TODO overflow-safe math?
-	gi.gasMeter.ConsumeGas(gi.gasConfig.ReadCostPerByte*sdk.Gas(len(value)), "ReadPerByte")
+	gs.gasMeter.ConsumeGas(gs.gasConfig.ReadCostPerByte*sdk.Gas(len(value)), sdk.GasReadPerByteDesc)
 
 	return value
 }
 
 // Implements KVStore.
-func (gi *gasKVStore) Set(key []byte, value []byte) {
-	gi.gasMeter.ConsumeGas(gi.gasConfig.WriteCostFlat, "WriteFlat")
+func (gs *gasKVStore) Set(key []byte, value []byte) {
+	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostFlat, sdk.GasWriteCostFlatDesc)
 	// TODO overflow-safe math?
-	gi.gasMeter.ConsumeGas(gi.gasConfig.WriteCostPerByte*sdk.Gas(len(value)), "WritePerByte")
-	gi.parent.Set(key, value)
+	gs.gasMeter.ConsumeGas(gs.gasConfig.WriteCostPerByte*sdk.Gas(len(value)), sdk.GasWritePerByteDesc)
+	gs.parent.Set(key, value)
 }
 
 // Implements KVStore.
-func (gi *gasKVStore) Has(key []byte) bool {
-	gi.gasMeter.ConsumeGas(gi.gasConfig.HasCost, "Has")
-	return gi.parent.Has(key)
+func (gs *gasKVStore) Has(key []byte) bool {
+	gs.gasMeter.ConsumeGas(gs.gasConfig.HasCost, sdk.GasHasDesc)
+	return gs.parent.Has(key)
 }
 
 // Implements KVStore.
-func (gi *gasKVStore) Delete(key []byte) {
+func (gs *gasKVStore) Delete(key []byte) {
 	// No gas costs for deletion
-	gi.parent.Delete(key)
+	gs.parent.Delete(key)
```

Comment:
I know you didn't change this, but \why is deleting 0 cost? Shouldn't it be just as complex as  `Has`?

Reply:
This is a good observation. We're freeing up space, but I agree it should charge the same price as `Has`. Thoughts @cwgoes ?

Reply:
We don't charge because space is freed - but actually this might be a DoS vector though, good catch. Let's charge the same as `Has`.